src/Servers/IIS/AspNetCoreModuleV2/InProcessRequestHandler/StartupExceptionApplication.h

```diff
@@ -15,32 +15,32 @@ class StartupExceptionApplication : public InProcessApplicationBase
         IHttpApplication& pApplication,
         HINSTANCE moduleInstance,
         BOOL disableLogs,
-        HRESULT hr)
+        HRESULT hr,
+        BYTE* errorPageContent,
+        int length)
         : m_disableLogs(disableLogs),
         m_HR(hr),
         m_moduleInstance(moduleInstance),
+        m_errorPageContent(errorPageContent),
+        m_length(length),
         InProcessApplicationBase(pServer, pApplication)
     {
     }
 
     ~StartupExceptionApplication() = default;
 
-    HRESULT CreateHandler(IHttpContext *pHttpContext, IREQUEST_HANDLER ** pRequestHandler)
+    HRESULT CreateHandler(IHttpContext* pHttpContext, IREQUEST_HANDLER** pRequestHandler)
     {
-        *pRequestHandler = new ServerErrorHandler(*pHttpContext, 500, 30, "Internal Server Error", m_HR, m_moduleInstance, m_disableLogs, IN_PROCESS_RH_STATIC_HTML);
-        return S_OK;
-    }
+        *pRequestHandler = new ServerErrorHandler(*pHttpContext, 500, 30, "Internal Server Error", m_HR, m_moduleInstance, m_disableLogs, IN_PROCESS_RH_STATIC_HTML, m_errorPageContent, m_length);
```

Comment:
Content-Type?

Reply:
It's set here: https://github.com/aspnet/AspNetCore/pull/8518/files#diff-1b6e0cba731b19c6281127415e6d3b0fR56. Think we should be setting charset=utf-8?